### PR TITLE
Add CredScan suppressions

### PR DESCRIPTION
**Pull request by MihaZupan**

### CODE REVIEW

File: .config/CredScanSuppressions.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+    "tool": "Credential Scanner",
+    "suppressions": [
+        {
+            "_justification": "Legitimate key/cert used for testing",
+            "file": [
+                "\\test\\TestCertificates\\testCert.pfx",
```
Comment (ManickaP) [inline .config/CredScanSuppressions.json:7]: Why backslashes? And why the starting backslash? BTW, would this work on Linux? And shouldn't we suppress the whole directory? Is it even possible `test/TestCertificates/*`?

File: .config/CredScanSuppressions.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            ]
+        },
+        {
+            "_justification": "Generic example password in documentation",
+            "file": "\\docs\\docfx\\articles\\proxyhttpclientconfig.md",
+            "placeholder": "1234abc"
+        },
+        {
+            "_justification": "Fake password used in test code",
+            "file": "\\test\\ReverseProxy.Tests\\Configuration\\CertificateConfigLoaderTests.cs",
```
Comment (ManickaP) [inline .config/CredScanSuppressions.json:25]: This is fragile, once someone changes the text it won't work. Could we suppress this inline, like: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs#L357

File: .config/CredScanSuppressions.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                "\\test\\TestCertificates\\https-ecdsa-protected.key",
+                "\\test\\TestCertificates\\https-ecdsa.key",
+                "\\test\\TestCertificates\\https-rsa-protected.key",
+                "\\test\\TestCertificates\\https-rsa.key"
+            ]
+        },
+        {
+            "_justification": "Generic example password in documentation",
+            "file": "\\docs\\docfx\\articles\\proxyhttpclientconfig.md",
+            "placeholder": "1234abc"
```
Comment (ManickaP) [inline .config/CredScanSuppressions.json:21]: Doesn't this line: https://github.com/microsoft/reverse-proxy/blob/master/docs/docfx/articles/proxyhttpclientconfig.md#L21 trigger the cred scan?
If not, than cool. If yes, we might exclude the whole file.

File: .config/CredScanSuppressions.json (hunk 1 above)
Comment (MihaZupan) [reply .config/CredScanSuppressions.json:7]: > Why backslashes? 

I started with https://github.com/dotnet/aspnetcore/blob/master/.config/CredScanSuppressions.json as the template.
I see we use forward slashes in runtime, I will update to match runtime.

> BTW, would this work on Linux?

I assume so if aspnetcore isn't seeing problems

> Is it even possible test/TestCertificates/*?

I tried using * or just the directory but that doesn't seem to work. Docs also just say `An absolute path or file name`.

File: .config/CredScanSuppressions.json (hunk 3 above)
Comment (MihaZupan) [reply .config/CredScanSuppressions.json:21]: Didn't trigger for me

File: .config/CredScanSuppressions.json (hunk 2 above)
Comment (MihaZupan) [reply .config/CredScanSuppressions.json]: Changed, though inline suppressions seem just as fragile to me.

**Review by ManickaP — APPROVED** [review]
LGTM, thanks!

**Pull request merged**